libs/iblprefilter/include/filament/iblprefilter/IBLPrefilterContext.h

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_IBL_PREFILTER_IBLPREFILTER_H
+#define TNT_IBL_PREFILTER_IBLPREFILTER_H
+
+#include <utils/Entity.h>
+
+#include <filament/Texture.h>
+
+namespace filament {
+class Engine;
+class View;
+class Scene;
+class Renderer;
+class Material;
+class MaterialInstance;
+class VertexBuffer;
+class IndexBuffer;
+class Camera;
+class Texture;
+} // namespace filament
+
+/**
+ * IBLPrefilterContext creates and initializes GPU state common to all environment map filters
+ * supported. Typically, only one instance per filament Engine of this object needs to exist.
+ *
+ * Usage Example:
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * #include <filament/Engine.h>
+ * using namespace filament;
+ *
+ * Engine* engine = Engine::create();
+ *
+ * IBLPrefilterContext context(engine);
+ * IBLPrefilterContext::SpecularFilter filter(context);
+ * Texture* texture = filter(environment_cubemap);
+ *
+ * IndirectLight* indirectLight = IndirectLight::Builder()
+ *     .reflections(texture)
+ *     .build(engine);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+class IBLPrefilterContext {
```

Comment:
Nice comment block!  If I had to prefilter several cubemaps, I guess it would be best to create only one context and one filter, then invoke the filter multiple times, right?

Reply:
Is there an alternate usage pattern that uses `createReflectionsTexture`, which could be mentioned here?

Reply:
> Nice comment block! If I had to prefilter several cubemaps, I guess it would be best to create only one context and one filter, then invoke the filter multiple times, right?

Yes, it would be a lot more efficient -- especially if you can reuse the same `SpecularFilter`.